Candidate: CVE-2024-23334
PublicDate: 2024-01-29 23:15:00 UTC
References:
 https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-23334
 https://github.com/aio-libs/aiohttp/security/advisories/GHSA-5h86-8mv2-jq9f
 https://github.com/aio-libs/aiohttp/pull/8079
 https://github.com/aio-libs/aiohttp/commit/1c335944d6a8b1298baf179b7c0b3069f10c514b
Description:
 aiohttp is an asynchronous HTTP client/server framework for asyncio and
 Python. When using aiohttp as a web server and configuring static routes,
 it is necessary to specify the root path for static files. Additionally,
 the option 'follow_symlinks' can be used to determine whether to follow
 symbolic links outside the static root directory. When 'follow_symlinks' is
 set to True, there is no validation to check if reading a file is within
 the root directory. This can lead to directory traversal vulnerabilities,
 resulting in unauthorized access to arbitrary files on the system, even
 when symlinks are not present. Disabling follow_symlinks and using a
 reverse proxy are encouraged mitigations. Version 3.9.2 fixes this issue.
Ubuntu-Description:
Notes:
Mitigation:
Bugs:
Priority: medium
Discovered-by:
Assigned-to:
CVSS:
 nvd: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N [7.5 HIGH]


Patches_python-aiohttp:
upstream_python-aiohttp: needs-triage
trusty_python-aiohttp: DNE
xenial_python-aiohttp: ignored (end of standard support)
esm-apps/xenial_python-aiohttp: needs-triage
bionic_python-aiohttp: ignored (end of standard support)
esm-apps/bionic_python-aiohttp: needs-triage
focal_python-aiohttp: needs-triage
esm-apps/focal_python-aiohttp: needs-triage
jammy_python-aiohttp: needs-triage
esm-apps/jammy_python-aiohttp: needs-triage
mantic_python-aiohttp: needs-triage
devel_python-aiohttp: needs-triage
